url8752fbdeb0_sixtam_args_tgz-pJ8-cz_mff_dpp_args_examples_ConstraintTestJ8/src/cz/mff/dpp/args/Parser.java

```diff
@@ -0,0 +1,547 @@
+
+			if (param != null && isOptionParameter(param)) {
+				params.add(tokens.pop());
+			} else {
+				List<String> defaults = OptionUtils.getDefaultValues(option);
+				if (defaults.size() > 1) {
+					params.add(defaults.get(0));
+
+					if (defaults.size() != 1) {
+						warning("Warning: Using only first default value for option '%s'",
+								option.name());
+					}
+				}
+			}
+
+		}
+
+		return params;
+
+	}
+
+	/**
+	 * Returns parameter list for an {@code Argument} from the parsed input
+	 * arguments.
+	 * 
+	 * @param accessible
+	 * @param argument
+	 * @return list of parameters for the argument
+	 * @throws ConfException
+	 *             if cannot be decided what parameters the argument should have
+	 * 
+	 * @see Argument
+	 */
+
+	private List<String> getArgumentParams(AccessibleObject accessible,
+			Argument argument) throws ConfException {
+
+		List<String> stringValues = new LinkedList<String>();
+
+		if (ReflectUtils.isSimpleType(accessible)) {
+
+			String stringValue;
+
+			stringValue = used.getArgument(argument.index());
+			stringValues.add(stringValue);
+
+		} else if (ReflectUtils.isArrayType(accessible)) {
+
+			int index = argument.index();
+			int size = argument.size();
+
+			stringValues = used.getArgument(index, size);
+
+		} else {
+			throw new ConfException("Cannot assign argument for '%s'",
+					accessible);
+		}
+
+		return stringValues;
+	}
+
+	// ------------------------------------------------------------------------
+	// AUXILIARY
+	// ------------------------------------------------------------------------
+	/**
+	 * 
+	 * Prepares environment and then sets objects with parameters.
+	 * 
+	 * 
+	 * @param accessible
+	 *            the object to assign parameters to
+	 * @param stringValues
+	 *            values for the object
+	 * @param annotation
+	 *            associated annotation of the object
+	 * @throws CheckException
+	 *             if the object does not pass some test
+	 * @throws ConfException
+	 *             if setting values of the object fails
+	 */
+	private void set(AccessibleObject accessible, List<String> stringValues,
+			Annotation annotation) throws CheckException, ConfException {
+
+		Class<?> factory = getFactoryType(annotation);
+		Class<?> valueType = ReflectUtils.getValueType(accessible);
+
+		if (factory != None.class) {
+			info("Registering factory class '%s' for '%s'", factory, valueType);
+			FactoryRegistry.register(valueType, factory);
+		}
+
+		Configurator.set(target, accessible, stringValues);
+
+		if (factory != None.class) {
+			info("Unregistring factory for '%s'", valueType);
+			FactoryRegistry.unregister(valueType);
+		}
+
+	}
+
+	/**
+	 * 
+	 * Checks the target object against specification declared by its
+	 * annotations.
+	 * 
+	 * 
+	 * @throws ParseException
+	 */
+	private void check() throws ParseException {
+		Inspector inspector = new Inspector(annotatedObjects, used);
+		try {
+			inspector.check();
+		} catch (CheckException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)